metagen_tools/tablegen.m

```diff
@@ -0,0 +1,130 @@
+
+function output = tablegen (basepath2)
+    % Get all the videos from the directory
+    file_struct = dir(fullfile(basepath2,'*.avi'));
+    %remove hidden files
+    file_struct = file_struct(~strncmp({file_struct.name}, '.', 1));
+    % Extract base name
+    bstring=char(basepath2);
+    File_name=bstring(1+max(strfind(basepath2,'\')):end); % ***** edit '\' or '/' depending on mac or windows
+    % Extract the timestamps from the filenames
+    timestamps = regexp({file_struct.name}, '-\d+', 'match');
+    timestamps = cellfun(@(x) extractAfter(x,'-'),timestamps,'UniformOutput',false);
+    timestamps = cellfun(@(x) str2double(x),timestamps,'UniformOutput',false);
+    [~, sortedIndices] = sort( [timestamps{:}]);
+    file_struct=file_struct(sortedIndices);
+    subid=[];
+    basepath=[];
+    genotype=[];
+    age=[];
+    session_date=[];
+    dob=[];
+    vidname=[];
+    basename=[];
+    exposure=[];
+    maze_width_cm=[];
+    maze_length_cm=[];
+    maze_type=[];
+    maze_color=[];
+    cue_position=[];
+    task_name=[];
+    condition=[];
+    treatment=[];
+    objects=[];
+    paradigm=[];
+    moved_object=[];
+    trial_start_1=[];
+    trial_stop_1=[];
+    trial_start_2=[];
+    trial_stop_2=[];
+    trial_start_3=[];
+    trial_stop_3=[];
+    trial_start_4=[];
+    trial_stop_4=[];
+    % none columns
+    none=[];
+    for i=1:length(sortedIndices)
+        %nones and constant columns
+        maze_length_cm=[maze_length_cm;39];
+        none=[none;"none"];
+        task_name=[task_name;"cpp"];
+        %subid
+        vid=file_struct(i).name;
+        subid=[subid;vid(1:4)];
+        %Basepath
+        pos=find(sortedIndices==i);
+        basepath=[basepath;string(basepath2)];
+        %Session Date
+        daytemp=extractBefore(file_struct(pos).date," "); 
+        date=datetime(daytemp, 'InputFormat', 'dd-MMM-yyyy');
+        date=datestr(date, 'mm/dd/yy');
+        session_date=[session_date;string(date)];
+        %video name
+        vidname=[vidname;string(vid)];
+        %base name
+        basename=[basename;string(File_name)];
+        %exposure and condition
+        
+        if contains(basename,'pretest')
```

Comment:
Make 68 - 104 adaptable for different behavioral paradigms. Dates on video files can be used for exposure calculation. Use recursive approach to get all videos from subject folder.